aegisthus-distcp/src/main/java/Distcp.java

```diff
@@ -228,68 +223,70 @@ protected void setReducers(Job job, int fileCount) {
 		}
 	}
 
-	protected int setupInput(Job job, Path inputPath, String[] inputFiles, String manifestPath) throws Exception {
+	protected int setupInput(Job job, Path inputPath, String[] inputFiles, String manifest) throws Exception {
+		if (manifest != null) {
+			Path manifestPath = new Path(manifest);
+			FileSystem fs = manifestPath.getFileSystem(job.getConfiguration());
+			BufferedReader br = new BufferedReader(new InputStreamReader(fs.open(manifestPath)));
+			ArrayList<String> manifestInputs = new ArrayList<String>();
+			String l;
+			while ((l = br.readLine()) != null) {
+				LOG.info(String.format("manifest inputfile: %s", l));
+				manifestInputs.add(l);
+			}
+			inputFiles = manifestInputs.toArray(new String[manifestInputs.size()]);
```

Comment:
it's a bit confusing we are repurposing `inputFiles` when `manifest` is not null, can we define a new variable (`inputFilesToCheck`) and set `inputFiles` to it when `manifest` is NULL, and go through this logic flow otherwise?